src/data/move-attrs/sacrificial-full-restore-attr.ts

```diff
@@ -26,12 +26,18 @@ export class SacrificialFullRestoreAttr extends SacrificialAttr {
   }
 
   override applyEffect(user: Pokemon, target: Pokemon, move: Move): boolean {
+    const party = user.getParty();
+
     // We don't know which party member will be chosen, so pick the highest max HP in the party
-    const maxPartyMemberHp = globalScene
-      .getPlayerParty()
+    const maxPartyMemberHp = party
       .map((p) => p.getMaxHp())
       .reduce((maxHp: number, hp: number) => Math.max(hp, maxHp), 0);
 
+    /**
+     * @todo If the incoming Pokemon does not get any HP healed, status healed, or PP restored,
+     * There should be a battler tag applied to the Pokemon which should expire whenever the heal
+     * would be needed
+     */
```

Comment:
```suggestion
    /**
     * @todo If the incoming Pokemon does not get any HP healed, status healed, or PP restored,
     * There should be an arena tag applied to the field which should expire whenever the heal
     * would be needed
     */
```
nit: An arena tag would be more fitting since the stored heal targets a field position rather than a specific Pokemon